sdk/core/azure-core/src/main/java/com/azure/core/util/ReferenceManager.java

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import com.azure.core.util.implementation.ReferenceManagerImpl;
+
+/**
+ * This interface represents the managing references to {@link Object Objects} and providing the ability to run a
+ * cleaning operation once the object is no longer able to be reference.
```

Comment:
nit: hard to parse

Reply:
Words weren't done well here... will re-write this